File: codes/python/206.反转链表.py

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
	def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
		if not head:
			return None

		prev = None
		curr = head
		next = None

		while curr:
			next = curr.next
			curr.next = prev
			prev = curr
			curr = next

		return prev

File: codes/golang/206.反转链表.go

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {

	if head == nil {
		return nil
	}

	var prev, curr, next *ListNode = nil, head, nil

	for curr != nil {
		next = curr.Next
		curr.Next = prev
		prev = curr
		curr = next
	}

	return prev
}